openedx/core/djangoapps/util/signals.py

```diff
@@ -30,5 +30,5 @@ def _clear_request_cache(**kwargs):
     Once a celery task completes, clear the request cache to
     prevent memory leaks.
     """
-    if getattr(settings, 'CLEAR_REQUEST_CACHE_ON_TASK_COMPLETION', True) and not getattr(settings, 'CELERY_ALWAYS_EAGER', False):
+    if getattr(settings, 'CLEAR_REQUEST_CACHE_ON_TASK_COMPLETION', True):
```

Comment:
By using the CLEAR_REQUEST_CACHE_ON_TASK_COMPLETION setting to control whether the cache is cleared rather than two settings makes it easier to test. In tests we want to use CELERY_ALWAYS_EAGER to avoid a dependency on a separate worker, but we still want to test that the code effectively clears the cache when the celery task completes.